kbase-extension/static/kbase/js/widgets/function_output/kbaseConditionSetViewer.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * @public
+ */
+'use strict';
+
+define (
+	[
+		'kbwidget',
+		'bootstrap',
+		'jquery',
+		'narrativeConfig',
+		'kbaseAuthenticatedWidget',
+		'jquery-dataTables',
+		'knhx',
+		'widgetMaxWidthCorrection'
+	], function(
+		KBWidget,
+		bootstrap,
+		$,
+		Config,
+		kbaseAuthenticatedWidget,
+		jquery_dataTables,
+		knhx,
+		widgetMaxWidthCorrection
+	) {
+    return KBWidget({
+        name: 'kbaseConditionSet',
+        parent : kbaseAuthenticatedWidget,
+        version: '0.0.1',
+        options: {
+            obj_ref: null,
+            wsURL: Config.url('workspace'),
+            loadingImage: Config.get('loading_gif')
+        },
+
+        init: function(options) {
+            this._super(options);
+
+            this.options.obj_ref = this.options.upas.obj_ref;
+
+            this.$messagePane = $("<div/>").addClass("kbwidget-message-pane kbwidget-hide-message");
+            this.$elem.append(this.$messagePane);
+
+            this.$mainPanel = $("<div>").addClass("").hide();
+            this.$elem.append(this.$mainPanel);
+
+            if (!this.options.obj_ref) {
+                this.renderError("No ConditionSet to render!");
+            } else if (!this.options.kbCache && !this.authToken()) {
+                this.renderError("No cache given, and not logged in!");
+            } else {
+                this.token = this.authToken();
+                this.render();
+            }
+
+            return this;
+        },
+
+        render: function() {
+            this.ws = new Workspace(this.options.wsURL, {token: this.token});
+            this.loading(false);
+            this.$mainPanel.hide();
+            this.$mainPanel.empty();
+            this.loadConditionSet();
+        },
+
+        conditions: {},
+        factors: {},
+
+        loadConditionSet: function() {
+            var self = this;
+            self.ws.get_objects([{ref:self.options.obj_ref}],
```

Comment:
should use `get_objects2({objects: [{ref: self.options.obj_ref}]})`

Reply:
eventually, this call should probably be wrapped in a Promise, but this is okay for the first pass.